### DISCUSSION TITLE
Generating a FSM at runtime

### DISCUSSION THREAD

**Discussion by brandonchinn178** [statelyai/xstate#4646] [Q&A]
In my application, I ask the user for the number of players, and then start the game. As a simplified explanation, each player is represented as a state and can transition to the next player. So I need to generate a machine at runtime with the number of states corresponding to the number of players.

When I do so, it works perfectly, except now I can't persist the machine, since "An inline child actor cannot be persisted". I understand why that's true generally (https://github.com/statelyai/xstate/issues/4410), since "Actor" is just an interface on top of a transition function, and you can't persist a function. But for my specific use case where the child actor is a state machine, `childActor.getPersistedSnapshot()` **does** work. Is there some other reason why an actor backed by a generated state machine can't be persisted?

Of course, perhaps I'm running into issues because this isn't how I'm supposed to use xstate. Is there a better / more idiomatic way to model this?

---

A simplified example:
```ts
createMachine({
  initial: 'setup',
  states: {
    setup: {
      invoke: {
        src: createMachine({
          // ...
          output: ({ context }) => ({ numPlayers: context.numPlayers }),
        }),
        id: 'setup',
        onDone: 'game',
      },
    },
    game: {
      entry: [
        assign({
          gameActor: ({ event, spawn }) => {
            const { numPlayers } = event.output

            const normalRounds = {}
            for (const i = 0; i < numPlayers; i++) {
              normalRounds[`round${i}`] = {
                on: {
                  NEXT: {
                    target: i + 1 < numPlayers ? `round${i + 1}` : 'done',
                  },
                },
              }
            }

            const gameMachine = createMachine({
              initial: 'round0',
              states: {
                ...normalRounds,
                done: {
                  type: 'final',
                },
              },
            })

            return spawn(gameMachine, { id: 'game' })
          },
        }),
      ],
      exit: stop('game'),
    },
  },
})
```

**Comment by christophe-g** [comment]
@brandonchinn178 - did you find a workaround for this ?

**Reply by brandonchinn178** [reply]
Nope

**Reply by christophe-g** [reply]
Thanks for the answer.

For what it's worth, I am just seeing the undocumented `__unsafeAllowInlineActors` option parameter

`game.getPersistedSnapshot({__unsafeAllowInlineActors: true})` should work, but you would need to handle `children.game.src`  (which is now the full StateMachine instance) in your own implementation of `persistedSnapshotLogic` ...
